# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/iankitthakur/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


My lane is Refresh / Content Opportunity Scoring: finding which content pages are losing search visibility and are worth a human review. I chose it because I can see its output directly (a ranked list of pages with a reason and an action for each), it fits the data the internship gives me (daily search clicks, impressions and position per page), and it lets me compare a simple readable rule with a learned model on the same pages.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Question:** which pages should a content editor review first, because their impressions are likely to fall?

**Decision it improves:** how an editor spends limited review time. A team can open only a limited number of pages each week, so the order matters.

**Who acts:** a content editor or SEO lead, who reads the ranked list, opens each page, and decides whether to change it.

**Cost of a wrong call:** a false alarm wastes review time (I assume about 30 minutes per page), and a change made to a page that did not need one can introduce a problem. Missing a page that really is declining means the problem is noticed later. Because of this, every recommendation goes through a person and the list is decision-support, not an automatic action.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


In [10]:
import pandas as pd

url = "https://raw.githubusercontent.com/iankitthakur/flyrank-ml-internship/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)
print("pages in the starter slice:", len(df), "| columns:", df.shape[1])

down = df["trend_direction"].str.lower().eq("down")
print("pages whose impressions went down:", int(down.sum()), f"({round(down.mean() * 100, 1)}%)")
print("median change in impressions among pages with a value (%):", round(df["trend_pct"].median(), 1))
df["trend_direction"].value_counts()

pages in the starter slice: 30000 | columns: 44
pages whose impressions went down: 16262 (54.2%)
median change in impressions among pages with a value (%): -33.5


,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


Three numbers from the starter slice (observed): it holds 30,000 pages; 54.2% of them had their impressions go down between the previous and the last 30 days; and among pages with a measured change, the median change was about −33.5%. More than half the pages are declining, so an editor who can review only a few dozen pages needs a sensible order for them. That makes the lane worth the next seven weeks.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**What my work can say:**
- Observed: patterns in this sample of pages, for example how many pages declined and how clicks and position relate.
- Measured: how a rule or a model ranks pages compared with a baseline, on the same split, with the base rate shown.
- Directional: signals that tend to go together with decline in this data, without a claim of cause.
- Decision-support: a ranked list for a human to review, with reason codes.

**What it will never say:**
- That an edit or refresh caused or will cause traffic to recover.
- That I know how Google ranks pages, or that I can predict Google.
- That results from one month and one sample apply to every site.
- Anything about a named client, domain or private query.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.